assets/js/components/OverlayNotification/AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification.js

```diff
@@ -0,0 +1,236 @@
+import { useFeature } from '../../hooks/useFeature';
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../modules/analytics-4/datastore/constants';
+import { getContextScrollTop } from '../../util/scroll';
+import OverlayNotification from './OverlayNotification';
+import { MODULES_ADSENSE } from '../../modules/adsense/datastore/constants';
+
+const { useSelect, useDispatch } = Data;
+
+export const ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION =
+	'AnalyticsAndAdSenseLinkedOverlayNotification';
+
+export default function AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification() {
+	const ga4AdSenseIntegration = useFeature( 'ga4AdSenseIntegration' );
+	const breakpoint = useBreakpoint();
+
+	const isShowingNotification = useSelect( ( select ) =>
+		select( CORE_UI ).isShowingOverlayNotification(
+			ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const { setOverlayNotificationToShow, dismissOverlayNotification } =
+		useDispatch( CORE_UI );
+
+	const isDismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed(
+			ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const isDismissing = useSelect( ( select ) =>
+		select( CORE_USER ).isDismissingItem(
+			ANALYTICS_ADSENSE_LINKED_OVERLAY_NOTIFICATION
+		)
+	);
+
+	const analyticsModuleConnected = useSelect( ( select ) => {
+		if ( isDismissed ) {
+			return null;
+		}
+		return select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+	} );
+
+	const adSenseModuleConnected = useSelect( ( select ) => {
+		if ( isDismissed ) {
+			return null;
+		}
+		return select( CORE_MODULES ).isModuleConnected( 'adsense' );
+	} );
+
+	const canViewSharedAnalytics = useSelect( ( select ) => {
+		if ( isDismissed ) {
+			return null;
+		}
+		return select( CORE_USER ).hasAccessToShareableModule( 'analytics-4' );
+	} );
+	const canViewSharedAdSense = useSelect( ( select ) => {
+		if ( isDismissed ) {
+			return null;
+		}
+		return select( CORE_USER ).hasAccessToShareableModule( 'adsense' );
+	} );
+
+	const isAdSenseLinked = useSelect( ( select ) => {
+		if ( isDismissed ) {
+			return null;
+		}
+
+		return select( MODULES_ANALYTICS_4 ).getAdSenseLinked();
+	} );
+
+	const adSenseAccountID = useSelect( ( select ) => {
+		if ( adSenseModuleConnected ) {
+			return select( MODULES_ADSENSE ).getAccountID();
+		}
+
+		return null;
+	} );
+
+	const { startDate, endDate } = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} )
+	);
+
+	const args = {
+		startDate,
+		endDate,
+		dimensions: [ 'pagePath', 'adSourceName' ],
+		metrics: [ { name: 'totalAdRevenue' } ],
+		filter: {
+			fieldName: 'adSourceName',
+			stringFilter: {
+				matchType: 'EXACT',
+				value: `Google AdSense account (${ adSenseAccountID })`,
+			},
+		},
+		orderby: [ { metric: { metricName: 'totalAdRevenue' }, desc: true } ],
+		limit: 1,
+	};
+
+	const data = useSelect( ( select ) => {
+		if (
+			! isDismissed &&
```

Comment:
It is better to also explicitly check if `isDismissed` `false` here before deciding whether to make a report request. To avoid report going out while `isDismissed` is `undefined`